### DISCUSSION TITLE
Graph Generation

### DISCUSSION THREAD

**Discussion by jcendava** [Green-Software-Foundation/hack#79] [Ask Anything] [answered]
In the [IF documentation](https://if.greensoftware.foundation/specification/manifest-file#an-executable-impact-calculation-manifest), it suggests the SDK and/or some tooling can be used to generated more complex graphs. Is there any additional guidance on this?

> Manifest files will be able to represent simple calculation manifest but to handle larger, more complex systems, we will have to write Graphs as code using our SDK.

More specifically, can graphs (child > children) be modified at runtime, i.e. by a pipeline plugin?

Thanks
James

Duplicate of https://github.com/Green-Software-Foundation/if/discussions/368

**Comment by jmcook1186** [comment, marked as answer]
Hi - the url you provided links to a page that was removed quite a while ago. There's no tooling that can programmatically generate trees in the manifest for you at the moment. The only plugin we have that manipulates the tree structure is `groupby` but that just reorganizes existing outputs into new groups according to the keys you provide in config.
